src/Arcus.Messaging.Pumps.ServiceBus.KeyRotation/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,61 @@
+    /// </summary>
+    // ReSharper disable once InconsistentNaming
+    public static class IServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds a background job to the <see cref="IServiceCollection"/> to automatically restart a <see cref="AzureServiceBusMessagePump"/> with a specific <paramref name="jobId"/>
+        /// when the Azure Key Vault secret that holds the Azure Service Bus connection string was updated.
+        /// </summary>
+        /// <param name="services">The collection of services to add the job to.</param>
+        /// <param name="jobId">The unique background job ID to identify which message pump to restart.</param>
+        /// <param name="subscriptionNamePrefix">The name of the Azure Service Bus subscription that will be created to receive <see cref="CloudEvent"/>'s.</param>
+        /// <param name="serviceBusTopicConnectionStringSecretKey">The secret key that points to the Azure Service Bus Topic connection string.</param>
+        /// <param name="targetConnectionStringKey">The secret key where the connection string credentials are located for the target message pump that needs to be auto-restarted.</param>
+        /// <exception cref="ArgumentNullException">
+        ///     Thrown when the <paramref name="services"/> or the searched for <see cref="AzureServiceBusMessagePump"/> based on the given <paramref name="jobId"/> is <c>null</c>.
+        /// </exception>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="subscriptionNamePrefix"/> or <paramref name="serviceBusTopicConnectionStringSecretKey"/> is blank.</exception>
+        public static IServiceCollection WithAutoRestartOnRotatedCredentials(
+            this IServiceCollection services,
+            string jobId,
+            string subscriptionNamePrefix,
+            string serviceBusTopicConnectionStringSecretKey,
+            string targetConnectionStringKey)
+        {
+            Guard.NotNull(services, nameof(services), "Requires a collection of services to add the re-authentication background job");
+            Guard.NotNullOrWhitespace(subscriptionNamePrefix, nameof(subscriptionNamePrefix), "Requires a non-blank subscription name of the Azure Service Bus Topic subscription, to receive Azure Key Vault events");
+            Guard.NotNullOrWhitespace(serviceBusTopicConnectionStringSecretKey, nameof(serviceBusTopicConnectionStringSecretKey), "Requires a non-blank secret key that points to a Azure Service Bus Topic");
+            Guard.NotNullOrWhitespace(targetConnectionStringKey, nameof(targetConnectionStringKey), "Requires a non-blank secret key that points to the credentials that holds the connection string of the target message pump");
+
+            services.AddCloudEventBackgroundJob(subscriptionNamePrefix, serviceBusTopicConnectionStringSecretKey);
+            services.WithServiceBusMessageHandler<ReAuthenticateOnRotatedCredentialsMessageHandler, CloudEvent>(serviceProvider =>
```

Comment:
Are we sure this doesn't impact our other message pump?

This might be another argument for https://github.com/arcus-azure/arcus.messaging/issues/152 to ensure that.

Reply:
When other background jobs or message pumps are receiving cloud events, yes, this message handler will be one of the registered ones, but 1) it filters on secret new version created events (and with the message body filter we can even stricken it further), and 2) it uses the ID of the message pump to make sure that only the correct pump is restarted.
There exists a problem though, when multiple pumps/jobs are receiving cloud events and want to auto restart. Then, the consumer should stricken the cloud event filter, I guess.

Reply:
Or, we should have a way to add the job ID to the messaging context, but that would mean we should have access to how the cloud events background job create messaging contexts.

Reply:
But yes, when we do #152 we can make sure that we have a separate `IServiceCollection` for each message router.

Reply:
Let's keep it as is and #152 will fix it.